From 1c1a261448c8b3c2162b81a5eceb4f252b6ac642 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Thu, 28 Mar 2013 14:15:26 +0800
Subject: [PATCH 436/502] ENGR00255885 fix UVC camera impact sound recorder
 working problem.

The root cause is closing device node twice and close system input
output device handle. It cause GM open file and can not operate the
file in Sound Recorder.
The solution is when close device node, make sure it is larger than
0 and make the device handle to be -1.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 mx6/libcamera2/DeviceAdapter.cpp | 5 ++++-
 mx6/libcamera2/UvcDevice.cpp     | 6 ++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/mx6/libcamera2/DeviceAdapter.cpp b/mx6/libcamera2/DeviceAdapter.cpp
index afcadb8..2ae3a26 100755
--- a/mx6/libcamera2/DeviceAdapter.cpp
+++ b/mx6/libcamera2/DeviceAdapter.cpp
@@ -59,7 +59,10 @@ DeviceAdapter::DeviceAdapter()
 DeviceAdapter::~DeviceAdapter()
 {
     // Close the camera handle and free the video info structure
-    close(mCameraHandle);
+    if (mCameraHandle > 0) {
+        close(mCameraHandle);
+        mCameraHandle = -1;
+    }
 
     if (mVideoInfo) {
         delete mVideoInfo;
diff --git a/mx6/libcamera2/UvcDevice.cpp b/mx6/libcamera2/UvcDevice.cpp
index e80ae9b..37a38a6 100644
--- a/mx6/libcamera2/UvcDevice.cpp
+++ b/mx6/libcamera2/UvcDevice.cpp
@@ -471,8 +471,10 @@ status_t UvcDevice::stopDeviceLocked()
         }
     }
 
-    close(mCameraHandle);
-    mCameraHandle = 0;
+    if (mCameraHandle > 0) {
+        close(mCameraHandle);
+        mCameraHandle = -1;
+    }
     return ret;
 }
 
-- 
1.8.0

